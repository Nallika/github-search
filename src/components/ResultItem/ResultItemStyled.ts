import styled from 'styled-components';

const Wrapper = styled.div`
  padding-top: 5px;
`;

const Link = styled.a`
  font-size: large;
  font-weight: bold;
  padding-top: 2px;
`;

const Text = styled.p`
  font-size: medium;
  padding: 0;
  margin: 0;
`

export { Wrapper, Link, Text }
