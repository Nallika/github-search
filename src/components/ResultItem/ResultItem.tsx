import React from 'react';

import {SearchItemData} from '../../types';
import { Wrapper, Link, Text } from './ResultItemStyled';

/**
 * Display individual search results element
 * @constructor
 */
const ResultItem: React.FC<SearchItemData> = (data) => (
  <Wrapper>
    <Link href={data.url}>{data.name}</Link>
    <Text>{`Stars: ${data.stargazerCount}`}</Text>
    <Text>{`Forks: ${data.forkCount}`}</Text>
  </Wrapper>
);


export default ResultItem;
