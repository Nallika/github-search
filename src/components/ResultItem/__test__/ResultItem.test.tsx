import React from 'react';
import { render } from '@testing-library/react';
import ResultItem from '../ResultItem';

test('`<ResultItem />` Should render component', () => {
  const mockData = {
    id: 'string',
    forkCount: 100,
    name: 'string',
    stargazerCount: 100,
    url: 'string'
  }
  // @ts-ignore
  const { container } = render(<ResultItem data={mockData} />);

  expect(container).toMatchSnapshot();
});
