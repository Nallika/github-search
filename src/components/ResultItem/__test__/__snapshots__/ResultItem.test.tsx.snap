// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<ResultItem />\` Should render component 1`] = `
<div>
  <div
    class="sc-bdfCDU gqhEQV"
  >
    <a
      class="sc-gsTDqH gseWtg"
    />
    <p
      class="sc-dlfmHC bAmRbQ"
    >
      Stars: undefined
    </p>
    <p
      class="sc-dlfmHC bAmRbQ"
    >
      Forks: undefined
    </p>
  </div>
</div>
`;
